docarray/array/array/io.py

```diff
@@ -291,6 +296,69 @@ def to_json(self) -> str:
         """
         return json.dumps([doc.json() for doc in self])
 
+    @classmethod
+    def from_csv(cls, file_path: str, encoding: str = 'utf-8') -> 'DocumentArray':
+        """
+        Load a DocumentArray from a csv file following the schema defined in the
+        :attr:`~docarray.DocumentArray.document_type` attribute.
+        The column names have to match the fields of the document type.
+        For nested fields use dot-separated access paths, such as 'image__url'.
+
+        :param file_path: path to csv file to load DocumentArray from.
+        :param encoding: encoding used to read the csv file. Defaults to 'utf-8'.
+        :return: DocumentArray
+        """
+        from docarray import DocumentArray
+
+        doc_type: Type[BaseDocument] = cls.document_type
+        if doc_type == AnyDocument:
+            raise TypeError(
+                'There is no document schema defined. '
+                'To load from csv, please specify the DocumentArray\'s document type.'
+            )
+
+        da = DocumentArray[doc_type]()  # type: ignore
+        with open(file_path, 'r', encoding=encoding) as fp:
+            lines = csv.DictReader(fp, dialect='excel')
+            fields = lines.fieldnames
```

Comment:
What's the type of `lines` and `fields`? I think a type annotation would be nice here, would make it easier to understand the code further down below